twilight-lavalink/README.md

```diff
@@ -13,13 +13,32 @@ handle sending voice channel updates to Lavalink by processing events via
 the [client's `process` method][`Lavalink::process`], which you must call
 with every Voice State Update and Voice Server Update you receive.
 
+A breakdown of how this functions:
+- The client is [`Lavalink`](crate::client::Lavalink) that forwards the required events from Discord.
+    - We read the [Voice State and Voice Server Updates](https://discord.com/developers/docs/topics/gateway-events#voice) from discord to format the data to send to a Lavalink VoiceUpdate Event.
+    - There is a lower level [node](crate::node) that processes this for you. It isn't recommended to use this but rather the lavalink struct with the players. If you don't find functionality please open up and issue to expose what you need.
+- You send the client an [outgoing event](crate::model::outgoing). These include play, pause, seek, etc. You send these through the [player](crate::player) that is attached to Lavalink.
+- If you want to search or load you need to create a http client currently and then you can use [these helpers functions](crate::http#functions) to generate the http uri and body to send over your http client. you will then get response you can deserialize as json into the structs in the [http module](crate::http).
+
+***NOTE: We currently only support `v4` of Lavlink. Support for `v3` is dropped. There was big changes in the api meaning the outgoing are now using a http client instead of websockets. The json request and responses all changed naming and fields changed.***
+
+Currently some [Filters](crate::model::outgoing::Filters) are not yet supported. There are some unsupported end points that were added yet such as [Lavalink Info](https://lavalink.dev/api/rest.html#get-lavalink-version) or [Session Api](https://lavalink.dev/api/rest.html#session-api) that weren't previously available. If you would like native support for something please reach out and open an issue for that feature. The porting only ported the functionality of the previous `v3` forward.
+
 ## Features
 
 ### `http-support`
```

Comment:
Do we want to unconditionally make `http` public now that it's reached 1.0? Also the name is unclear if it's the HTTP protocol or the http crate that's supported. It should probably be reamed, making users read the docs is a failure of naming.